libdec/arch/riscv.js

```diff
@@ -188,8 +190,10 @@
             },
             lui: function(instr) {
                 var dst = instr.parsed.opd[0];
-                var imm20 = instr.parsed.opd[1] << 12;
-                return Base.assign(dst, '0x' + imm20.toString(16)) ;
+                var imm = parseInt(instr.parsed.opd[1]);
+                var imm20 = imm << 12;
+                var n = new Long(imm20, ((Global.evars.archbits > 32) && imm & 0x80000) ? -1 : 0, true);
+                return Base.assign(dst, '0x' + n.toString(16)) ;
```

Comment:
Are you trying to sign-extend?

Reply:
Yes. Forgive naive code. With current HEAD I get:
```
    lui a4, 0xff880                   |     a4 = 0xffffffffffffffff;
    addi a4, a4, -1
```
which is kinda annoying. I tried to fix it by various C-like means, but I always got `0` at imm20 stage, so I had to resort to throwing new Long object to make it simply work.